src/framework/constraints.ml

```diff
@@ -675,7 +675,7 @@ struct
       | Entry f        -> tf_entry var edge prev_node f
       | Ret (r,fd)     -> tf_ret var edge prev_node r fd
       | Test (p,b)     -> tf_test var edge prev_node p b
-      | ASM (_, _, _)  -> tf_asm var edge prev_node (* TODO: use ASM fields for something? *)
+      | ASM _          -> tf_asm var edge prev_node (* TODO: use ASM fields for something? *)
```

Comment:
The relevant data should be taken from the `ASM` constructor and passed as actual arguments instead of looking at `ctx.edge` in the transfer function implementation.